plugins/bigquery-api-backend/src/service/router.ts

```diff
@@ -37,21 +37,31 @@ export async function createRouter(
   const readAppConfig = config.getConfig('bigqueryapi');
   const projectId = readAppConfig.getString('projectId');
   const gbdataset = readAppConfig.getString('gbdataSet');
+  const { GoogleAuth, Impersonated } = require('google-auth-library');
+
 
   router.get('/dataset', async (_, response) => {
 
-    const { JWT } = require('google-auth-library');
-    const keys = require('../config/cedar-setup.json');
-    const client = new JWT({
-      email: keys.client_email,
-      key: keys.private_key,
-      scopes: ['https://www.googleapis.com/auth/bigquery'],
+    const scopes = 'https://www.googleapis.com/auth/cloud-platform'
+    const auth = new GoogleAuth({
+      scopes: scopes
+    });
+    const client = await auth.getClient();
+    let targetPrincipal = 'googlebigquery@cedar-setxxxxxx.iam.gserviceaccount.com'
```

Comment:
we can change the service account in the app-config in the future

Reply:
Its five minutes of work, but intentionally kept inside to avoid compromise (if over permissioned)